features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -78,3 +78,125 @@ Feature: SDN compoment upgrade testing
+    Then the step should succeed
+
+    Given the AllowNamespaceAndPod policy is applied to the "policy-upgrade1" namespace
+    Then the step should succeed
+
+    When I use the "policy-upgrade1" project
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail 
+    And the output should not contain "Hello"
+    When I execute on the "<%= cb.pod3 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail
+    And the output should not contain "Hello"
+
+    When I use the "policy-upgrade2" project
+    When I execute on the "<%= cb.pod4 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should succeed
+    And the output should contain "Hello"
+    When I execute on the "<%= cb.pod5 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail 
+    And the output should not contain "Hello"
+    When I execute on the "<%= cb.pod4 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod3ip %>:8080 |
+    Then the step should succeed 
+    And the output should contain "Hello"
+
+  # @author asood@redhat.com
+  # @case_id OCP-38751
+  @admin
+  @upgrade-check
+  Scenario: Check the networkpolicy works well after upgrade
+    Given I switch to cluster admin pseudo user
+    When I use the "policy-upgrade1" project
+    Given status becomes :running of 2 pods labeled:
+      | name=test-pods |
+    And evaluation of `pod(1).ip_url` is stored in the :pod2ip clipboard
+    And evaluation of `pod(0).name` is stored in the :pod1 clipboard
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail
+    Given a pod becomes ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(2).name` is stored in the :pod3 clipboard
+    And evaluation of `pod(2).ip_url` is stored in the :pod3ip clipboard
+    When I execute on the "<%= cb.pod3 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail
+
+    When I use the "policy-upgrade2" project
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(3).name` is stored in the :pod4 clipboard
+    When I execute on the "<%= cb.pod4 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(4).name` is stored in the :pod5 clipboard
+    When I execute on the "<%= cb.pod5 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail 
+    When I execute on the "<%= cb.pod4 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod3ip %>:8080 |
+    Then the step should succeed 
```

Comment:
I think here should be failed.  pod4 should not be able to access pod3ip since pod3 do not have label with 'name=test-pods'.
I'm not sure if run again this case after updating. Could you help paste the logs ?

Reply:
pod3ip is of pod3 in policy-upgrade1 namespace and has label name=hello-idle. Network policy does not apply to that pod it only applies to pod with label name=test-pods. pod4 has label name=test-pods. This is the only policy in namespace policy-upgrade1. I believe there has to be deny all policy if you want completely isolate namespaces and then allow some access.

When I execute on the "<%= cb.pod4 %>" pod:                                                     # features/step_definitions/pod.rb:216
      [11:59:57] INFO> Shell Commands: oc exec test-rc-sspx2  --kubeconfig=/home/asood/workdir/asood-asood/ocp4_admin.kubeconfig -n policy-upgrade2  -i -- curl -s --connect-timeout 5 10.128.3.112:8080
      Hello Pod!
      
      STDERR:
      E0325 07:59:57.618183  280520 v2.go:105] read /dev/stdin: resource temporarily unavailable
      
      [11:59:58] INFO> Exit Status: 0
      | curl | -s | --connect-timeout | 5 | <%= cb.pod3ip %>:8080 |
    Then the step should succeed                                                                    # features/step_definitions/common.rb:4
    And the output should contain "Hello"                                                           # features/step_definitions/common.rb:113